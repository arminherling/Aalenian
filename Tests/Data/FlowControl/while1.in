def func() 
{ 
    while true
    {
        return
    }

    return
}
