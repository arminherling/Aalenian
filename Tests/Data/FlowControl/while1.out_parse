FunctionDefinitionStatement: {
    NameExpression: func
    ParametersNode(0): {
    }
    BlockNode(2): {
        WhileStatement: {
            Condition: {
                BoolLiteral: true
            }
            BlockNode(1): {
                ReturnStatement: {
                }
            }
        }
        ReturnStatement: {
        }
    }
}
