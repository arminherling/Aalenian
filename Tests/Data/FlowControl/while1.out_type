TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->()
    Name: func
    Parameters(0): {
    }
    ReturnType: {
        Type: void
    }
    Body(2): {
        TypedWhileStatement: {
            Condition: {
                BoolValue: {
                    Type: bool
                    Value: true
                }
            }
            Body(1): {
                TypedReturnStatement: {
                    Type: void
                    Expression: {
                    }
                }
            }
        }
        TypedReturnStatement: {
            Type: void
            Expression: {
            }
        }
    }
}
