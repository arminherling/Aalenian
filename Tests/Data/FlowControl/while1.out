FunctionDefinitionStatement: {
    Name: func
    Parameters(0): {
    }
    Block(2): {
        WhileStatement: {
            Condition: {
                Bool: true
            }
            Block(1): {
                ReturnStatement: {
                }
            }
        }
        ReturnStatement: {
        }
    }
}
